examples/with-segment-analytics/components/analytics.tsx

```diff
@@ -0,0 +1,27 @@
+'use client'
+
+import { useEffect } from 'react'
+import { usePathname, useSearchParams } from 'next/navigation'
+import { AnalyticsBrowser } from '@segment/analytics-next'
+
+// This write key is associated with https://segment.com/nextjs-example/sources/nextjs.
+const DEFAULT_WRITE_KEY = 'NPsk1GimHq09s7egCUlv7D0tqtUAU5wa'
+
+const analytics = AnalyticsBrowser.load({
+  writeKey: process.env.NEXT_PUBLIC_SEGMENT_WRITE_KEY || DEFAULT_WRITE_KEY,
+})
+
+export function useSegment() {
+  return analytics;
```

Comment:
Hey, sorry! this isn't what I meant -- you had it right before.

I meant encapsulating just  the page call with a hook!

Reply:
I updated it to use a hook.